### DISCUSSION TITLE
Overlay Text or Images

### DISCUSSION THREAD

**Discussion by vedeht** [tiny-pilot/tinypilot#1844] [Ideas]
I've been trying to figure out a way to run some text overlay on the display.  I want it to show up to the hdmi monitor output.  Any ideas if this is possible or not?

**Comment by db39** [comment]
Hi @vedeht, thanks for your question.

I'm not sure I understand exactly what you mean by overlaying images/text on the display. TinyPilot captures your PC's display and shows it in a web browser, allowing you to control your PC without connecting a physical keyboard, mouse, and display.

Could you clarify what you mean and share an example of what you'd like to achieve?

**Reply by vedeht** [reply]
If I wanted to output some text to the HDMI monitor such as the Server Name or something.

**Reply by vedeht** [reply]
I also have some software that can display relevant information to what is displayed on the monitor.  I want the ability to put in some text such as closed captioning information or labels.  I know it can be done with a standard display but what I want is the ability to do it on the kvm itself.

**Reply by db39** [reply]
Thanks for those details. There isn't a way to create a text overlay on top of TinyPilot's display (such as the server name), unfortunately. Some users change their TinyPilot's hostname to match the target machine their TinyPilot connects to. For example, if I had a server with the name `server1` I could change my TinyPilot's hostname to `TinyPilot-server1` and check my browser's address bar if I needed to check which TinyPilot I'm using.

Please let me know if you have any questions.

**Reply by vedeht** [reply]
I am trying to eliminate the need to buy an HDMI video overlay device.  I am sure I can figure it out.  What is the device is the tinypilot registering as the output?  Maybe I can manipulate the output to the hdmi?

**Reply by db39** [reply]
>What is the device is the tinypilot registering as the output? Maybe I can manipulate the output to the hdmi?

I'm not sure that's possible. TinyPilot uses `/dev/video0` for its capture device.

>I am trying to eliminate the need to buy an HDMI video overlay device.

An HDMI video overlay device might be the quickest and simplest way to handle this use case. However, if you're technical, you could create your own overlay for TinyPilot's web interface. We have a guide for getting started contributing to the project [here](https://github.com/tiny-pilot/tinypilot/blob/master/CONTRIBUTING.md).

I hope that helps! Please let me know if you have any other questions.

**Reply by vedeht** [reply]
I'll definitely figure it out.  Just answer me this, video comes in on /dev/video0 how does it output to the hdmi to the physical monitor?  I know it uses uStreamer and Janus.  Does the output also get piped to the onboard hdmi output?  I'm trying to figure out where to start to attempt to do it.  I just need pointed into the right direction and i'll make it work.

**Reply by db39** [reply]
>Does the output also get piped to the onboard hdmi output?

No, TinyPilot doesn't output the captured image to the onboard HDMI. The only output you'll see on the onboard HDMI output if you connect a physical display is the lite Raspberry Pi terminal.

**Reply by vedeht** [reply]
How does TinyPilot output the captured HDMI input to the physical monitor that is connected to the TinyPilot HDMI output?

**Reply by db39** [reply]
>How does TinyPilot output the captured HDMI input to the physical monitor that is connected to the TinyPilot HDMI output?

TinyPilot doesn't output the captured image anywhere other than the TinyPilot web interface. TinyPilot isn't designed to have a physical monitor connected to any of its ports.
